### ISSUE TITLE
When using CanvasKit, Gallery web benchmarks fail

### ISSUE THREAD

**Issue by pennzht** [flutter/flutter#59082]
## Background

While running web benchmark tests for the new gallery under CanvasKit (https://github.com/flutter/flutter/pull/57576), an error was thrown, and the benchmark could not complete normally.

This does **not** happen when the benchmark tests are run without CanvasKit.

## Steps to Reproduce

1. Clone https://github.com/pennzht/flutter/tree/gallery-benchmarks.ref
2. Checkout commit `037e84f6824994292da0639183a8dec4e546a38e`
3. Navigate to the directory `dev/devicelab`
4. Run `dart bin/run.dart -t web_benchmarks_canvaskit`

**Expected results:** Benchmark finishes without error.

**Actual results:** The following error was thrown:
```
[web_benchmarks_canvaskit] [STDOUT] [Gallery] Warming up.
[web_benchmarks_canvaskit] [STDOUT] [Gallery] Warm-up finished.
[web_benchmarks_canvaskit] [STDOUT] [Gallery] ==== List of demos to be run ====
[web_benchmarks_canvaskit] [STDOUT] [Gallery] shrine@study
[web_benchmarks_canvaskit] [STDOUT] [Gallery] rally@study
[web_benchmarks_canvaskit] [STDOUT] [Gallery] crane@study
[web_benchmarks_canvaskit] [STDOUT] [Gallery] fortnightly@study
[web_benchmarks_canvaskit] [STDOUT] [Gallery] starter@study
[web_benchmarks_canvaskit] [STDOUT] [Gallery] ==== End of list of demos to be run ====
[web_benchmarks_canvaskit] [STDOUT] [Gallery] Running demo "shrine@study"
[web_benchmarks_canvaskit] [STDOUT] [Gallery] Running demo "rally@study"
[web_benchmarks_canvaskit] [STDERR] Task failed: Error: Cannot perform %TypedArray%.prototype.set on a detached ArrayBuffer
[web_benchmarks_canvaskit] [STDERR] TypeError: Cannot perform %TypedArray%.prototype.set on a detached ArrayBuffer
[web_benchmarks_canvaskit] [STDERR]     at Uint8Array.set (<anonymous>)
[web_benchmarks_canvaskit] [STDERR]     at h (https://unpkg.com/canvaskit-wasm@0.16.1/bin/canvaskit.js:17:452)
[web_benchmarks_canvaskit] [STDERR]     at Function.a.SkFontMgr.FromData (https://unpkg.com/canvaskit-wasm@0.16.1/bin/canvaskit.js:64:331)
[web_benchmarks_canvaskit] [STDERR]     at JsFunction.callMethod$2 (http://localhost:9999/main.dart.js:78902:42)
[web_benchmarks_canvaskit] [STDERR]     at http://localhost:9999/main.dart.js:60022:74
[web_benchmarks_canvaskit] [STDERR]     at _wrapJsFunctionForAsync_closure.$protected (http://localhost:9999/main.dart.js:7197:15)
[web_benchmarks_canvaskit] [STDERR]     at _wrapJsFunctionForAsync_closure.call$2 (http://localhost:9999/main.dart.js:70333:12)
[web_benchmarks_canvaskit] [STDERR]     at _awaitOnObject_closure.call$1 (http://localhost:9999/main.dart.js:70319:32)
[web_benchmarks_canvaskit] [STDERR]     at Object._rootRunUnary (http://localhost:9999/main.dart.js:7667:18)
[web_benchmarks_canvaskit] [STDERR]     at StaticClosure.<anonymous> (http://localhost:9999/main.dart.js:207468:16)
[web_benchmarks_canvaskit] [STDERR] 
[web_benchmarks_canvaskit] [STDERR] Stack trace:
[web_benchmarks_canvaskit] [STDERR] dart:async                                                 _Completer.completeError
[web_benchmarks_canvaskit] [STDERR] package:flutter_devicelab/tasks/web_benchmarks.dart 87:23  runWebBenchmark.<fn>.<fn>
[web_benchmarks_canvaskit] [STDERR] ===== asynchronous gap ===========================
[web_benchmarks_canvaskit] [STDERR] dart:async                                                 _asyncThenWrapperHelper
[web_benchmarks_canvaskit] [STDERR] package:flutter_devicelab/tasks/web_benchmarks.dart        runWebBenchmark.<fn>.<fn>
[web_benchmarks_canvaskit] [STDERR] package:shelf/src/cascade.dart 72:52                       Cascade.handler.<fn>
[web_benchmarks_canvaskit] [STDERR] package:shelf/src/cascade.dart 73:47                       Cascade.handler.<fn>.<fn>
[web_benchmarks_canvaskit] [STDERR] dart:async                                                 new Future.sync
[web_benchmarks_canvaskit] [STDERR] package:shelf/src/cascade.dart 73:21                       Cascade.handler.<fn>
[web_benchmarks_canvaskit] [STDERR] package:shelf/shelf_io.dart 101:29                         handleRequest
[web_benchmarks_canvaskit] [STDERR] package:shelf/shelf_io.dart 63:34                          serveRequests.<fn>.<fn>
[web_benchmarks_canvaskit] [STDERR] ===== asynchronous gap ===========================
[web_benchmarks_canvaskit] [STDERR] dart:_http                                                 _HttpServer.listen
[web_benchmarks_canvaskit] [STDERR] package:shelf/shelf_io.dart 63:14                          serveRequests.<fn>
[web_benchmarks_canvaskit] [STDERR] package:shelf/src/util.dart 24:20                          catchTopLevelErrors
[web_benchmarks_canvaskit] [STDERR] package:shelf/shelf_io.dart 62:3                           serveRequests
[web_benchmarks_canvaskit] [STDERR] package:flutter_devicelab/tasks/web_benchmarks.dart 123:7  runWebBenchmark.<fn>
[web_benchmarks_canvaskit] [STDERR] ===== asynchronous gap ===========================
[web_benchmarks_canvaskit] [STDERR] dart:async                                                 _asyncThenWrapperHelper
[web_benchmarks_canvaskit] [STDERR] package:flutter_devicelab/tasks/web_benchmarks.dart        runWebBenchmark.<fn>
[web_benchmarks_canvaskit] [STDERR] package:flutter_devicelab/framework/utils.dart 463:24      inDirectory
[web_benchmarks_canvaskit] [STDERR] package:flutter_devicelab/tasks/web_benchmarks.dart 28:16  runWebBenchmark
[web_benchmarks_canvaskit] [STDERR] bin/tasks/web_benchmarks_canvaskit.dart 13:18              main.<fn>
[web_benchmarks_canvaskit] [STDERR] package:flutter_devicelab/framework/framework.dart 170:36  _TaskRunner._performTask.<fn>
[web_benchmarks_canvaskit] [STDERR] package:stack_trace                                        Chain.capture
[web_benchmarks_canvaskit] [STDERR] package:flutter_devicelab/framework/framework.dart 169:11  _TaskRunner._performTask
[web_benchmarks_canvaskit] [STDERR] package:flutter_devicelab/framework/framework.dart 101:41  _TaskRunner.run
[web_benchmarks_canvaskit] [STDERR]
```

<details>
  <summary>Logs</summary>

(`flutter analyze` finds no issues within `dev/benchmarks/macrobenchmarks/`.)

Xcode has an issue, but these benchmarks are **web** benchmarks and do not depend on Xcode.

```
[✓] Flutter (Channel unknown, 1.19.0-1.0.pre.103, on Mac OS X 10.14.6 18G2022, locale en-DE)
    • Flutter version 1.19.0-1.0.pre.103 at /Users/tianguang/Documents/f/flutter
    • Framework revision 037e84f682 (22 hours ago), 2020-06-08 21:56:48 +0200
    • Engine revision dfdd88deb7
    • Dart version 2.9.0 (build 2.9.0-13.0.dev 87c5e3612b)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/tianguang/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 11.2.1)
    • Xcode at /Applications/Xcode 11.2.1.app/Contents/Developer
    • Xcode 11.2.1, Build version 11B500
    ! CocoaPods 1.7.5 out of date (1.8.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart
        side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade:
        sudo gem install cocoapods

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio
    • Android Studio at /Applications/Android Studio with Blaze.app/Contents
    • Flutter plugin version 45.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] Connected device (4 available)
    • Android SDK built for x86 • emulator-5554 • android-x86    • Android 10 (API 29) (emulator)
    • macOS                     • macOS         • darwin-x64     • Mac OS X 10.14.6 18G2022
    • Web Server                • web-server    • web-javascript • Flutter Tools
    • Chrome                    • chrome        • web-javascript • Google Chrome 83.0.4103.97

! Doctor found issues in 1 category.
```

</details>

**Comment by yjbanov** [comment]
This issue depends on our ability to run gallery benchmarks in the devicelab, which is currently being reworked (see https://github.com/flutter/flutter/pull/66351). We can't start work on this issue until those bits land, so I'm putting this issue back in the queue.

**Comment by guidezpl** [comment]
With respect to https://github.com/flutter/gallery/issues/1072, should this issue be closed?

**Comment by parlough** [comment]
I wasn't able to reproduce this specific issue after fixing a few others, so I'll close for now.

If anyone experiences this or a similar issue, please open a new issue with reproduction information for your case. Thanks!
